src/uct/api/uct_def.h

```diff
@@ -299,6 +299,38 @@ typedef ucs_status_t (*uct_sockaddr_conn_request_callback_t)(void *arg,
                                                              size_t length);
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Callback to fill the user's private data on the client side.
+ *
+ * This callback routine will be invoked on the client side before sending the
+ * transport's connection request to the server.
+ * It should be set by the client side when creating an endpoint.
```

Comment:
This sentence has some ambiguous language that makes it slow to parse. Instead of "it should be set by the client side when...", I think it is more clear to say "The callback routine must be set by the client when..."